com.microsoft.mrtk.uxcore/Keyboard/NonNative/NonNativeFunctionKey.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UX
+{
+    /// <summary>
+    /// Class representing a function key in the non native keyboard
+    /// </summary>
+    public class NonNativeFunctionKey : NonNativeKey
+    {
+        /// <summary>
+        /// Possible functionalities for a function key.
+        /// </summary>
+        public enum Function
+        {
+            // Commands
+            Enter,
+            Tab,
+            ABC,
+            Symbol,
+            Close,
+
+            // Editing
+            Shift,
+            CapsLock,
+            Space,
+            Backspace,
+
+            UNDEFINED = 255,
+        }
+
+        /// <summary>
+        /// The function of this key.
+        /// </summary>
+        [field: SerializeField, Tooltip("The function of this key.")]
+        public Function KeyFunction { get; private set; } = Function.UNDEFINED;
+
+        /// <inheritdoc/>
+        protected override void FireKey()
+        {
+            NonNativeKeyboard.Instance.ProcessFunctionKeyPress(this);
```

Comment:
Is there a reason why this passes the instance of the key instead of just the `KeyFunction`?

See other comment about possibly making extensible/breaking-change-proof `EventArgs` containers

Reply:
In case the user needs to know more about the key (e.g. access the GameObject, etc.)

Reply:
If this is the case, then all of the keys should pass a reference to themselves. I don't see why some keys should pass a reference to themselves, and others shouldn't

Reply:
The same applies to `NonNativeValueKey` as well.